pkg/controller/atlasproject/teams_test.go

```diff
@@ -437,4 +445,78 @@ func TestUpdateTeamState(t *testing.T) {
 		k8sClient.Get(context.Background(), types.NamespacedName{Name: team.ObjectMeta.Name, Namespace: team.ObjectMeta.Namespace}, team)
 		assert.Equal(t, 1, len(team.Status.Projects))
 	})
+
+	t.Run("must remove a team from Atlas is a team is unassigned", func(t *testing.T) {
+		logger := zaptest.NewLogger(t).Sugar()
+		workflowCtx := &workflow.Context{
+			Context: context.Background(),
+			Log:     logger,
+		}
+		testScheme := runtime.NewScheme()
+		akov2.AddToScheme(testScheme)
+		corev1.AddToScheme(testScheme)
+		secret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "my-secret",
+			},
+			Data: map[string][]byte{
+				"orgId":         []byte("0987654321"),
+				"publicApiKey":  []byte("api-pub-key"),
+				"privateApiKey": []byte("api-priv-key"),
+			},
+			Type: "Opaque",
+		}
+		project := &akov2.AtlasProject{
+			Spec: akov2.AtlasProjectSpec{
+				Name: "projectName",
+				ConnectionSecret: &common.ResourceRefNamespaced{
+					Name: "my-secret",
+				},
+			},
+			Status: status.AtlasProjectStatus{
+				ID: "projectID",
+			},
+		}
+		team := &akov2.AtlasTeam{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "testTeam",
+				Namespace: "testNS",
+			},
+			Status: status.TeamStatus{
+				ID:       "testTeamStatus",
+				Projects: []status.TeamProject{},
+			},
+		}
+		teamsMock := &atlas.TeamsClientMock{
+			RemoveTeamFromOrganizationFunc: func(orgID string, teamID string) (*mongodbatlas.Response, error) {
+				return nil, nil
+			},
+			RemoveTeamFromOrganizationRequests: map[string]struct{}{},
+		}
+		atlasProvider := &atlas.TestProvider{
+			ClientFunc: func(secretRef *client.ObjectKey, log *zap.SugaredLogger) (*mongodbatlas.Client, string, error) {
+				return &mongodbatlas.Client{
+					Teams: teamsMock,
+				}, "0987654321", nil
+			},
+		}
+		k8sClient := fake.NewClientBuilder().
+			WithScheme(testScheme).
+			WithObjects(secret, project, team).
+			Build()
+		reconciler := &AtlasProjectReconciler{
+			Client:        k8sClient,
+			Log:           logger,
+			AtlasProvider: atlasProvider,
+		}
+		teamRef := &common.ResourceRefNamespaced{
+			Name:      team.Name,
+			Namespace: "testNS",
+		}
+
+		err := reconciler.updateTeamState(workflowCtx, project, teamRef, true)
```

Comment:
I am confused, this tests `updateTeamState ` while no changes have been done in this function in this contribution in the first place.

Reply:
Yes, but if verifies if it actually removes the team if it is unassigned from all AtlasProjects. See the new unit-test added.